### DISCUSSION TITLE
How to update the state after deleting an item

### DISCUSSION THREAD

**Discussion by ghost** [vercel/swr#945] [Q&A] [answered]
When not using SWR I would updated my set state like this once I had deleted an item for example

` setData(data.filter((d) => d._id !== id));`

But I am not really sure how to do this now. I am not using SWR for any mutations but using Axios directly like this:

```
  const handleDelete = async (id) => {
    try {
      const response = await Axios.delete(
        `${process.env.REACT_APP_BACKEND_URL}/v1/user/${id}`,
        {
          headers: {
            Authorization: "Bearer " + auth.token,
          },
        }
      );
      console.log(response);
      mutate( `${process.env.REACT_APP_BACKEND_URL}/v1/users`);
    } catch (err) {
      console.log(err);
    }
  };
```

I then have tried to use the mutate function to reload the users but this doesn't actually do anything and I don't see any errors. Clearly I am not using it the intended way. Any assistance would be appreciated.

**Comment by koba04** [comment, marked as answer]
How do you fetch users' data using `useSWR`?

**Reply by ghost** [reply]
I have a custom hook which I use like this:

```
  const { data, isError, isLoading } = useHttp(
    `${process.env.REACT_APP_BACKEND_URL}/v1/users`,
    auth.token
  );

```
Here is the hook

```
import axios from "axios";
import useSwr from "swr";

export const useHttp = (url, token) => {
  const fetcher = (url, token) =>
    axios
      .get(url, { headers: { Authorization: "Bearer " + token } })
      .then((res) => res.data);
  const { data, error } = useSwr(token ? [url, token] : null, fetcher);

  return {
    data: data,
    isLoading: !error && !data,
    isError: error,
  };
};

```

**Reply by koba04** [reply]
To revalidate the data, you have to pass the same key to the mutate function.
In this case, you have to pass `[url, token]` as the key like the following.

```js
mutate([`${process.env.REACT_APP_BACKEND_URL}/v1/users`, auth.key])
```

You can also use [a bound mutate function](https://swr.vercel.app/docs/mutation#bound-mutate).

**Reply by ghost** [reply]
Ahhhh, thank you! I was really scratching my head on this but that solved it. Thanks so much!

**Reply by koba04** [reply]
Great 👍 Could you change the status of the discussion to `Answered`. It would be helpful for developers who have the same question.

**Reply by ghost** [reply]
Done!
